docs/designs/master_redesign.md

```diff
@@ -0,0 +1,136 @@
+# Master Design
+
+This document deseribes an ElasticDL master design.
+
+## Motivation
+
+We plan to provide elastic training feature to customized AllReduce training codes and other deep learning frameworks. Currently the master is designed for ElasticDL's own training framework. This requires us to re-design the master.
+After this re-design, the master consists of a few modular components and only launches those required components when providing elastic training feature to others.
```

Comment:
launches various required components while providing elastic training feature in different scenarios.